tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1547 @@
+            "inet": [
+                {
+                    "address": "10.10.10.56",
+                    "broadcast": "10.10.10.255",
+                    "netmask": "255.255.252.0",
+                }
+            ],
+            "inet6": [
+                {
+                    "address": "fe80::e23f:49ff:fe85:6aaf",
+                    "prefixlen": "64",
+                    "scope": "link",
+                }
+            ],
+            "up": True,
+        },
+        "lo": {
+            "inet": [{"address": "127.0.0.1", "netmask": "255.0.0.0"}],
+            "inet6": [{"address": "::1", "prefixlen": "128", "scope": "host"}],
+            "up": True,
+        },
+    }
+
+
+@pytest.fixture(scope="module")
+def freebsd_interfaces_dict():
+    return {
+        "": {"up": False},
+        "em0": {
+            "hwaddr": "00:30:48:ff:ff:ff",
+            "inet": [
+                {
+                    "address": "10.10.10.250",
+                    "broadcast": "10.10.10.255",
+                    "netmask": "255.255.255.224",
+                },
+                {
+                    "address": "10.10.10.56",
+                    "broadcast": "10.10.10.63",
+                    "netmask": "255.255.255.192",
+                },
+            ],
+            "up": True,
+        },
+        "em1": {"hwaddr": "00:30:48:aa:aa:aa", "up": False},
+        "lo0": {
+            "inet": [{"address": "127.0.0.1", "netmask": "255.0.0.0"}],
+            "inet6": [
+                {"address": "fe80::1", "prefixlen": "64", "scope": "0x8"},
+                {"address": "::1", "prefixlen": "128", "scope": None},
+            ],
+            "up": True,
+        },
+        "plip0": {"up": False},
+        "tun0": {
+            "inet": [{"address": "10.12.0.1", "netmask": "255.255.255.255"}],
+            "up": True,
+        },
+    }
+
+
+def test_sanitize_host_ip():
+    ret = network.sanitize_host("10.1./2.$3")
+    assert ret == "10.1.2.3"
+
+
+def test_sanitize_host_name():
+    """
+    Should not remove the underscore
+    """
+    ret = network.sanitize_host("foo_bar")
+    assert ret == "foo_bar"
+
+
+def test_host_to_ips():
+    """
+    NOTE: When this test fails it's usually because the IP address has
+    changed. In these cases, we just need to update the IP address in the
+    assertion.
+    """
+
+    # pylint doesn't like the }[host] below, disable typecheck
+    # pylint: disable=all
+    def getaddrinfo_side_effect(host, *args):
+        try:
+            return {
+                "github.com": [
+                    (2, 1, 6, "", ("192.30.255.112", 0)),
+                    (2, 1, 6, "", ("192.30.255.113", 0)),
+                ],
+                "ipv6host.foo": [
+                    (socket.AF_INET6, 1, 6, "", ("2001:a71::1", 0, 0, 0)),
+                ],
+            }[host]
+        except KeyError:
+            raise socket.gaierror(-2, "Name or service not known")
+
+    # pylint: enable=all
```

Comment:
```suggestion

    def getaddrinfo_side_effect(host, *args):
        if host == "github.com":
            return [
                    (2, 1, 6, "", ("192.30.255.112", 0)),
                    (2, 1, 6, "", ("192.30.255.113", 0)),
            ]
        if host = "ipv6host.foo":
            return [
                    (socket.AF_INET6, 1, 6, "", ("2001:a71::1", 0, 0, 0)),
                ]
        raise socket.gaierror(-2, "Name or service not known")

```
There. Same outcome, no need to disable pylint, and, no need to create a dictionary to return a key of it and if there's no key, catch the exception, etc....

Reply:
getting tired of refactoring unittest to pytest, even when the test works.
just wondering if time would be better spent on other work :( , especially when not mentioned in previous reviews, even though I appreciate the effort.

Reply:
> Done, getting tired of refactoring unittest to pytest, even when the test works.

Migrating to pytest shouldn't be a matter of just copying the test and minimally adjusting the tests, one should cleanup, and make use of pytest features.

> just wondering if time would be better spent on other work :( , especially when not mentioned in previous reviews, even though I appreciate the effort.

This is work, next time your on the test module it will be so much cleaner.